src/cmake_modules/FastDownwardMacros.cmake

```diff
@@ -1,100 +1,76 @@
 include(CMakeParseArguments)
 
-macro(check_and_set_compiler_flag FLAG)
-    include(CheckCXXCompilerFlag)
-    check_cxx_compiler_flag( "${FLAG}" FLAG_FOUND )
-    if(NOT FLAG_FOUND)
-        message(FATAL_ERROR "${CMAKE_CXX_COMPILER} does not support ${FLAG}")
-    endif()
-    message("Flag '${FLAG}' set for '${CMAKE_CXX_COMPILER}'")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${FLAG}")
-endmacro()
 
-macro(fast_downward_set_compiler_flags)
-    # Note: on CMake >= 3.0 the compiler ID of Apple-provided clang is AppleClang.
-    # If we change the required CMake version from 2.8.3 to 3.0 or greater,
-    # we have to fix this.
-    if(CMAKE_COMPILER_IS_GNUCXX OR ${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang")
-        check_and_set_compiler_flag( "-std=c++20" )
-
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g")
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -pedantic -Wnon-virtual-dtor -Wfloat-conversion")
-
-        if (CMAKE_COMPILER_IS_GNUCXX
-            AND NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 12
-            AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 13)
-            ## We ignore the warning "restrict" because of a bug in GCC 12:
-            ## https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105651
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-restrict")
-        endif()
 
-        ## Configuration-specific flags
-        set(CMAKE_CXX_FLAGS_RELEASE "-O3 -DNDEBUG -fomit-frame-pointer")
-        set(CMAKE_CXX_FLAGS_DEBUG "-O3")
-        if(USE_GLIBCXX_DEBUG)
-            set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -D_GLIBCXX_DEBUG")
-        endif()
-        set(CMAKE_CXX_FLAGS_PROFILE "-O3 -pg")
-    elseif(MSVC)
-        check_and_set_compiler_flag( "/std:c++20" )
-
-        # Enable exceptions.
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /EHsc")
-
-        # Use warning level 4 (/W4).
-        # /Wall currently detects too many warnings outside of our code to be useful.
-        string(REPLACE "/W3" "/W4" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
-
-        # Disable warnings that currently trigger in the code until we fix them.
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4456") # declaration hides previous local declaration
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4458") # declaration hides class member
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4459") # declaration hides global declaration
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4244") # conversion with possible loss of data
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4267") # conversion from size_t to int with possible loss of data
-
-        # TODO: Configuration-specific flags. We currently rely on the fact that
-        # CMAKE_CXX_FLAGS_RELEASE and CMAKE_CXX_FLAGS_DEBUG get reasonable settings
-        # from cmake. This is the case for most build environments, but we have less
-        # control over the way the binary is created.
+add_library(cxx_options INTERFACE)
```

Comment:
The way this is currently called, the library is defined when including the file, which happens before defining the project for `search`. At this point the `LANGUAGES` are not set yet, which could lead to complications.

Reply:
seems to work fine.